### DISCUSSION TITLE
New chat finetuned model - togethercomputer/GPT-NeoXT-Chat-Base-20B

### DISCUSSION THREAD

**Discussion by wywywywy** [oobabooga/text-generation-webui#264] [General]
Together just released a new model finetuned from GPT-NeoX that can act like a chat bot aka ChatGPT.

https://huggingface.co/togethercomputer/GPT-NeoXT-Chat-Base-20B

I got it running on my 3090 in bf16, although the VRAM is maxed out. `python server.py --model GPT-NeoXT-Chat-Base-20B --bf16 --listen-port 7861 --cai-chat --no-stream`

Quality seems pretty good but obviously not as good as the real ChatGPT. I'm excited to see if they'll do another one based on LLaMa (probably not because of the licence but one can dream).

![image](https://user-images.githubusercontent.com/19997747/224564035-210189fb-7ffa-431a-bca9-a7be99c88d26.png)

**Comment by oobabooga** [comment]
Wow, that's nice! Thanks for the heads up, I had not heard of this model.

**Comment by wywywywy** [comment]
Forgot to mention that here's the online test app with a bunch of example questions & answers: https://huggingface.co/spaces/togethercomputer/OpenChatKit

**Comment by lxe** [comment]
This model is amazing. It's also apache2, so you can build commercial apps out of it

**Comment by ye7iaserag** [comment]
So this works out of the box? no need to do or add any code the UI?
Also is that model loadable in 4bit? or with offloading?

**Comment by wywywywy** [comment]
Only LLaMa (and maybe OPT) works in 4-bit.

EDIT: Although other people are working on getting GPTQ to work for other models. So hopefully we'll be able to use this in 4-bit very soon.

**Comment by oobabooga** [comment]
GPTQ is implemented for OPT, but I have not managed to get good results with quantizing GALACTICA (a variant of OPT) :(

**Comment by ye7iaserag** [comment]
Got it working on 3080ti using
```
python server.py --auto-devices --gpu-memory 9 --cai-chat --load-in-8bit --listen --listen-port 8888 --model=GPT-NeoXT-Chat-Base-20B
```
Can we get a preset for it? (with Pygmalion settings it just doesn't shut up once it starts generating tokens xD)
![Untitleda](https://user-images.githubusercontent.com/2270440/225789042-e30f203f-e4c7-4dd4-a07e-e5190cebae2d.png)

**Reply by ye7iaserag** [reply]
![Untitled2](https://user-images.githubusercontent.com/2270440/225792971-3108c51f-7ef9-4e14-b84a-0b08292e8561.png)
Same again and it OOMed after a "hey there" message, maybe I'll drop the 9gbs of vram to 8

**Comment by OWKenobi** [comment]
For my taste, the answer of llama-13b is even more chatGPT-like:

"Do mussels have muscles? Give a detailed answer."

"Yes, mussels are animals that have muscle tissue. They live in salt water and filter food from it through their gills."
